types/mixpanel-browser/index.d.ts

```diff
@@ -153,7 +154,8 @@ export interface Mixpanel {
     has_opted_in_tracking(options?: Partial<HasOptedInOutOptions>): boolean;
     has_opted_out_tracking(options?: Partial<HasOptedInOutOptions>): boolean;
     identify(unique_id?: string): any;
-    init(token: string, config?: Partial<Config>, name?: string): Mixpanel;
+    init(token: string, config: Partial<Config>, name: string): Mixpanel;
+    init(token: string, config?: Partial<Config>): undefined;
```

Comment:
@moreal thanks for submitting. I get what you're trying to achieve here with this PR (having the types match the exact returns) and maybe that is the right approach. However, after looking at [the source code for `init`](https://github.com/mixpanel/mixpanel-js/blob/master/src/mixpanel-core.js#L208), it seems that `name` is indeed required and it [will return `undefined` as your PR changes suggest (and console an error)](https://github.com/mixpanel/mixpanel-js/blob/master/src/mixpanel-core.js#L211) if `name` is not passed in.

However, instead of having TS handle this scenario, I think just forcing name to be required should be all that is needed, no?

```suggestion
```

Reply:
Yeah, it's right. I will remove the definition with tests related to it. Thanks!